### PR TITLE
Customizable Session Incoming and Outgoing CircularBuffer size

### PR DESCRIPTION
**Pull request by MarioArnt**
Hi, 

I needed to change these parameters and saw a TODO annotation in the current code 👍 

Here is a PR allowing to customize these size by giving an optional additional ConnectionOptions.

Your feedback is very welcome 😄 

Have a great day

### CODE REVIEW

File: lib/connection.js [hunk 1]
```diff
@@ -208,6 +208,7 @@ var Connection = function (options, container) {
             && options.servername === undefined && options.host !== undefined) {
             this.options.servername = options.host;
         }
+        this.buffer_size = options.buffer_size;
```
Comment (grs) [inline lib/connection.js:211]: I think the option name should be a little bit clearer, e.g. session_buffer_size or similar. Otherwise looks good!

Comment (MarioArnt) [reply lib/connection.js]: Thanks for your quick review

Totally agree, that's way more clear :100:

> There are only two hard things in Computer Science: cache invalidation and naming things.

Comment (MarioArnt) [reply lib/connection.js]: Done + added some unit tests

Comment (grs) [reply lib/connection.js]: > There are only two hard things in Computer Science: cache invalidation and naming things.
and off-by-one errors!

**Comment by grs** [comment]
Thank you @MarioArnt !

**Pull request merged**